app/services/resource_recommendation.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+class ResourceRecommendation
+  def initialize(moment)
+    @moment = moment
+    @moment_keywords = []
+    @matched_resources = []
+  end
+
+  def get_keywords(array)
+    array.each do |item|
+      @moment_keywords.push(item['name'].split,
+                            html_clean(item['description']).split)
+    end
+  end
+
+  def modify_keywords
+    @moment_keywords = @moment_keywords.flatten
+    @moment_keywords.each do |keyword|
+      keyword.gsub!(%r{([_@#!%()\-=;><,{}\~\[\]\./\?\"\*\^\$\+\-]+)}, '')
+    end
+    @moment_keywords = @moment_keywords.map(&:downcase)
+  end
+
+  def match_keywords
+    all_resources.each do |resource|
+      resource_tags = resource['tags'].map do |tag|
```

Comment:
When you `map` something and then flats it, you can replace `map` with `flat_map`

https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-flat_map